code/datums/components/tippable.dm

```diff
@@ -0,0 +1,187 @@
+			return
+	do_tip(tipped_mob, tipper)
+
+/*
+ * Actually tip over the mob, setting it to tipped.
+ * Also invoking any callbacks we have, with the tipper as the argument,
+ * and set a timer to right our self-right our tipped mob if we can.
+ *
+ * tipped_mob - the mob who was tipped
+ * tipper - the mob who tipped the tipped_mob
+ */
+/datum/component/tippable/proc/do_tip(mob/living/tipped_mob, mob/tipper)
+	if(QDELETED(tipped_mob))
+		return
+
+	to_chat(tipped_mob, span_userdanger("You are tipped over by [tipper]!"))
+	to_chat(tipper, span_warning("You tip over [tipped_mob]."))
+	tipped_mob.visible_message(span_warning("[tipper] tips over [tipped_mob]."), ignored_mobs = list(tipped_mob, tipper))
+
+	set_tipped_status(tipped_mob, TRUE)
+	post_tipped_callback?.Invoke(tipper)
+	if(self_right_time > 0)
+		addtimer(CALLBACK(src, .proc/right_self, tipped_mob), self_right_time)
+
+/*
+ * Try to untip a mob that has been tipped.
+ * After a do-after is completed, we untip the mob.
+ *
+ * tipped_mob - the mob who is tipped
+ * untipper - the mob who is untipping the tipped_mob
+ */
+/datum/component/tippable/proc/try_untip(mob/living/tipped_mob, mob/untipper)
+	if(untip_time > 0)
+		to_chat(untipper, span_notice("You begin righting [tipped_mob]..."))
+		to_chat(tipped_mob, span_notice("[untipper] begins righting you!"))
+		untipper.visible_message(span_notice("[untipper] begins righting [tipped_mob]."), ignored_mobs = list(tipped_mob, untipper))
+		if(!do_after(untipper, untip_time, target = tipped_mob))
+			to_chat(untipper, span_warning("You fail to right [tipped_mob]."))
+			return
+
+	do_untip(tipped_mob, untipper)
+
+/*
+ * Actually untip over the mob, setting it to untipped.
+ * Also invoke any untip callbacks we have, with the untipper as the argument.
+ *
+ * tipped_mob - the mob who was tipped
+ * tipper - the mob who tipped the tipped_mob
+ */
+/datum/component/tippable/proc/do_untip(mob/living/tipped_mob, mob/untipper)
+	if(QDELETED(tipped_mob))
+		return
+
+	to_chat(tipped_mob, span_notice("You are righted by [untipper]!"))
+	to_chat(untipper, span_notice("You right [tipped_mob]."))
+	tipped_mob.visible_message(span_notice("[untipper] rights [tipped_mob]."), ignored_mobs = list(tipped_mob, untipper))
+
+	set_tipped_status(tipped_mob, FALSE)
+	post_untipped_callback?.Invoke(untipper)
+
+/*
+ * Proc called after a timer to have a tipped mob un-tip itself after a certain length of time.
+ * Sets our mob to untipped and invokes the untipped callback without any arguments if we have one.
+ *
+ * tipped_mob - the mob who was tipped, and is freeing itself
+ */
+/datum/component/tippable/proc/right_self(mob/living/tipped_mob)
+	if(!is_tipped || QDELETED(tipped_mob))
+		return
+
+	set_tipped_status(tipped_mob, FALSE)
+	post_untipped_callback?.Invoke()
+
+	to_chat(tipped_mob, span_notice("You right yourself."))
+	tipped_mob.visible_message(span_notice("[tipped_mob] rights itself."), ignored_mobs = tipped_mob)
```

Comment:
```suggestion
	tipped_mob.visible_message(
		span_notice("[tipped_mob] rights itself."),
		span_notice("You right yourself."),
		ignored_mobs = tipped_mob
		)
```
The `itself` part here could be substituted for more proper pronouns, but that'd require a feature of its own.